### PR TITLE
Nomenclature plot fixes

### PR DESCRIPTION
**Pull request by gzotti**
### Description

This introduces an option to encircle craters and satellite features (which are usually also craters) with ellipses.

I also just slightly simplified the code.
 

Fixes # (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**:
* Operating system: Windows 11
* Graphics Card: Geforce 3070Ti
* Qt 6.5.1

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by 10110111** [comment]
When I enable craters outline, I get the following repeated messages in the log:
```
Distance greater 1
Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
Distance greater 1
Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
Distance greater 1
Distance greater 1
Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

Distance greater 1
Distance greater 1
Distance greater 1
/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)

/home/ruslan/src/stellarium/src/core/modules/AtmosphereShowMySky.cpp:789: OpenGL error: 1281 (GL_INVALID_VALUE)
```

**Comment by gzotti** [comment]
> Hmm... It's interesting feature, but IMHO ellipses have troubles for scale factor:...

Not sure, do you mean display scaling? Planet scale factor (like 4x for the moon) works for me.

**Comment by gzotti** [comment]
I also see this in master:

> D:\StelDev\GIT\stellarium\src\core\StelApp.cpp:910: OpenGL error: 1281 (GL_INVALID_VALUE)

**Comment by 10110111** [comment]
> I also see this in master

Constantly repeating?

**Comment by gzotti** [comment]
Yes. I only have excluded a few plugins from building. 

>$ git log
>commit a79ca0497544eb076e1eab86fbf825e5554a103b (HEAD -> master, origin/master)

**Comment by alex-w** [comment]
> > Hmm... It's interesting feature, but IMHO ellipses have troubles for scale factor:...
> 
> Not sure, do you mean display scaling? Planet scale factor (like 4x for the moon) works for me.

Please show screenshot with outlines, but without scaling the Moon

**Comment by gzotti** [comment]
stellarium-093

**Comment by alex-w** [comment]
> stellarium-093

See screenshot from macOS (HiDPI monitor, device pixel ratio: 2):
stellarium-005

**Comment by alex-w** [comment]
OK, I see pixel ratio correction is applied twice - one time in `StelPainter::drawEllipse()` method and one time in `NomenclatureItem::draw()` method (see obtaining `screenRadius` by the fact). Please see tentative fix.

Plus I see flashing outlines for craters near lunar limb :(

**Comment by 10110111** [comment]
Yes, with this fix it's now better.

The option to outline craters need to be disabled when nomenclature is off, for consistency with other options that get disabled.

**Comment by alex-w** [comment]
If tentative fix is acceptable, then only flashes near limb need to be fixed

**Comment by gzotti** [comment]
Thanks for the hints. The flashing is better, but it seems some flicker remains due to rounding effects.

**Comment by gzotti** [comment]
I still have a problem to define the rotation angle in antiquity.

**Comment by gzotti** [comment]
I now exclude all ellipses which would extend over the limb. I think it's the best possible solution with that approach. Mare Orientale and other possible "large tangent disks" will hopefully always be excluded with that.

**Comment by alex-w** [comment]
Probably GUI option should be renamed now (Outline circles?) and tooltip with short description for this option really need.

**Review by alex-w — APPROVED** [review]

File: src/core/modules/NomenclatureMgr.cpp [hunk 1]
```diff
@@ -558,6 +554,18 @@ int NomenclatureMgr::getTerminatorMaxAltitude() const
 	return NomenclatureItem::terminatorMaxAltitude;
 }
 
+void NomenclatureMgr::setFlagOutlineCraters(bool b)
+{
+	NomenclatureItem::flagOutlineCraters = b;
+	conf->setValue("astro/flag_planets_nomenclature_outline_craters", b);
```
Comment (10110111) [inline src/core/modules/NomenclatureMgr.cpp:560]: This immediate saving is inconsistent with the rest of the options here (except the terminator ones, which are also inconsistent with the rest).

Comment (alex-w) [reply src/core/modules/NomenclatureMgr.cpp:560]: Ops... I agree

Comment (gzotti) [reply src/core/modules/NomenclatureMgr.cpp:560]: I know. I like the immediate storage of details much better than having to remember whether I have made any other detail changes before I save all options. Immediate here are color, terminator, solar altitudes, and now circles. The global (not-immediate) option is nomenclature or not.  But indeed I would like to have many more immediate-store settings.

Comment (10110111) [reply src/core/modules/NomenclatureMgr.cpp:560]: This inconsistency is very confusing.

Comment (alex-w) [reply src/core/modules/NomenclatureMgr.cpp:560]: Immediate storage of details is very bad for educational purposes (esp. for using Stellarium in classes) - it's good for personal uses only. Please move saving option into ConfigurationDialog class.

Or, at least, we should add support profiles - #439

**Review by alex-w — APPROVED** [review]

**Pull request merged**